src/gt4py/next/iterator/transforms/collapse_tuple.py

```diff
@@ -35,40 +105,77 @@ class CollapseTuple(eve.NodeTranslator):
       - `tuple_get(i, make_tuple(e_0, e_1, ..., e_i, ..., e_N))` -> `e_i`
     """
 
+    class Flag(enum.IntEnum):
```

Comment:
Did you look too long at Fortran namelists? 😜
This pass screams for pattern rewrite. Maybe we can do something in that direction here? Instead of packing all cases in the class, do more something that allows to pass a list of tuple of condition and (local) transformation?